### PR TITLE
Improve `FilterErrorBuilder.can_build` heuristics

### PR DESCRIPTION
**Pull request by pederhan**
Declares a `DEFAULT_OFFSET` value we can compare against. Decorates `find_word_with_char_offset` with `@cache` to let us calculate the offsets, check offsets, then call the builder again, without having to recalculate the offsets.

### CODE REVIEW

File: mreg_cli/errorbuilder.py [hunk 1]
```diff
@@ -107,12 +108,16 @@ def can_build(self) -> bool:  # noqa: D102
         return True
 
 
+DEFAULT_OFFSET = (-1, -1)
```
Comment (terjekv) [inline mreg_cli/errorbuilder.py:111]: Nitpicky, but is this really a "default_offset" or a "word_not_found_result"?

**Review by terjekv — APPROVED** [review]
LGTM

File: mreg_cli/errorbuilder.py (hunk 1 above)
Comment (pederhan) [reply mreg_cli/errorbuilder.py]: You are right that it is more of a sentinel value - albeit one that still contains valid indices. I have refactored it to be more clear now.

Comment (terjekv) [reply mreg_cli/errorbuilder.py]: The refactored code looks good! Thanks!

**Pull request merged**